mirgecom/boundary.py

```diff
@@ -38,91 +40,264 @@
+            self._viscous_flux_func = self._viscous_flux_for_prescribed_state
+        if not self._bnd_grad_cv_func:
+            self._bnd_grad_cv_func = self._identical_grad_cv
+        if not self._bnd_grad_temperature_func:
+            self._bnd_grad_temperature_func = self._identical_grad_temperature
 
     def _boundary_quantity(self, discr, btag, quantity, **kwargs):
         """Get a boundary quantity on local boundary, or projected to "all_faces"."""
+        from grudge.dof_desc import as_dofdesc
+        btag = as_dofdesc(btag)
         if "local" in kwargs:
             if kwargs["local"]:
                 return quantity
         return discr.project(btag, btag.with_dtag("all_faces"), quantity)
 
-    def inviscid_divergence_flux(self, discr, btag, gas_model, state_minus,
-                                 **kwargs):
+    def _boundary_state_pair(self, discr, btag, gas_model, state_minus, **kwargs):
+        return TracePair(btag,
+                         interior=state_minus,
+                         exterior=self._bnd_state_func(discr=discr, btag=btag,
+                                                       gas_model=gas_model,
+                                                       state_minus=state_minus,
+                                                       **kwargs))
+
+    def _temperature_for_prescribed_state(self, discr, btag,
+                                          gas_model, state_minus, **kwargs):
+        boundary_state = self._bnd_state_func(discr=discr, btag=btag,
+                                              gas_model=gas_model,
+                                              state_minus=state_minus,
+                                              **kwargs)
+        return boundary_state.temperature
+
+    def _identical_state(self, state_minus, **kwargs):
+        return state_minus
+
+    def _identical_grad_cv(self, grad_cv_minus, **kwargs):
+        return grad_cv_minus
+
+    def _identical_grad_temperature(self, grad_t_minus, **kwargs):
+        return grad_t_minus
+
+    def _gradient_flux_for_prescribed_cv(self, discr, btag, gas_model, state_minus,
+                                         **kwargs):
+        # Use prescribed external state and gradient numerical flux function
+        boundary_state = self._bnd_state_func(discr=discr, btag=btag,
+                                              gas_model=gas_model,
+                                              state_minus=state_minus,
+                                              **kwargs)
+        cv_pair = TracePair(btag,
+                            interior=state_minus.cv,
+                            exterior=boundary_state.cv)
+
+        actx = state_minus.array_context
+        nhat = thaw(discr.normal(btag), actx)
+        return self._boundary_quantity(
+            discr, btag=btag,
+            quantity=self._grad_num_flux_func(cv_pair, nhat), **kwargs)
+
+    def _gradient_flux_for_prescribed_temperature(self, discr, btag, gas_model,
```

Comment:
Comment on what "for prescribed temperature" means here. (Docstring/comment)

Reply:
This function will be eliminated on BC refactor.   I will comment it up to try to make it more clear in the interim.

Reply:
([8344799](https://github.com/illinois-ceesd/mirgecom/pull/581/commits/8344799ab04e877403b0e5162d2f6f9ece3505b4))